Hstreamdb 0.9

savonarola opened the pull request.

alissa-tung reviewed (changes requested):
Could you please rename ordering keys to partition keys? Since the semantics have changed.

alissa-tung commented on src/hstreamdb_ordkey_mgr.erl:
```diff
@@ -0,0 +1,92 @@
+    Req = #{'streamName' => Stream},
+    Options = #{channel => Channel, timeout => GRPCTimeout},
+    <<IntHash:128/big-unsigned-integer>> = crypto:hash(md5, OrderingKey),
+    case ?HSTREAMDB_CLIENT:list_shards(Req, Options) of
```
Generally, we should avoid `list_shards` RPC every time, we can just call it during producer init.

savonarola replied:
✅

savonarola commented:
Hi!

> Could you please rename ordering keys to partition keys? Since the semantics have changed.

✅ 

* I renamed the keys;
* Introduced a parameter for shards cache timeout (it can be set to a large value if shard list isn't supposed to change);
* Introduced shard-wise buffering. This allows to have PartitionKey set of any cardinality without memory penalty.

alissa-tung commented:
> Hi!
> 
> > Could you please rename ordering keys to partition keys? Since the semantics have changed.
> 
> ✅
> 
>     * I renamed the keys;
> 
>     * Introduced a parameter for shard cache timeout (it can be set to a large value if shard list isn't supposed to change);
> 
>     * Introduced shard-wise buffering. This allows to have PartitionKey set of any cardinality without memory penalty.

nice work!  LGTM

alissa-tung commented:
Hi, I am wondering where was the part of the list of records get serialized to batch records
